Diagnostic/diagnostic.py

```diff
@@ -695,6 +695,26 @@ def start_daemon():
         hutil.error("wait daemon start time out")
 
 
+def check_suspected_memory_leak(pid):
+    memory_leak_threshold_in_KB = 2000000  # Roughly 2GB
```

Comment:
The only place I can think of is the extension's publicSettings (JSON), but I don't think that can be used anytime soon. I think this issue can be revisited later if/when we truly want to make this feature a publicly advertised one (which I personally don't prefer). Please note that this is just a first step in trying to self-mitigate rare occasions.